Cargo.toml

```diff
@@ -11,6 +11,7 @@ anyhow = "1.0.41"
 combine = "4.6.0"
 midir = "0.7.0"
 rand = "0.8.4"
+spin_sleep = "1.0.0"
```

Comment:
Based on some reading it seemed like [`spin_sleep`](https://docs.rs/spin_sleep/1.0.0/spin_sleep/) was a good choice for a thread accurately sleeping for a given amount of time